File: Patterns/code/java/DesignPatterns/src/main/java/decorator/TomatoDecorator.java
package decorator;

class TomatoDecorator extends PizzaDecorator {
    public TomatoDecorator(Pizza pizza) {
        super(pizza);
    }

    @Override
    public void bake() {
        super.bake();
        System.out.println("Adding tomatoes");
    }
}

File: code/typescript/src/decorator/TomatoDecorator.ts
import { PizzaDecorator } from "./PizzaDecorator";
import { Pizza } from "./Pizza";

export class TomatoDecorator extends PizzaDecorator {
    constructor(pizza: Pizza) {
        super(pizza);
    }

    bake(): void {
        super.bake();
        console.log("Adding tomatoes");
    }
}
